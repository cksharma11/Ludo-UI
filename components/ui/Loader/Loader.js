import React from 'react';
import LoaderStyles from './Loder.style';

const Loader = () => {
  return (
    <div className="loader">
      <style jsx>{LoaderStyles}</style>
    </div>
  );
};

export default Loader;
