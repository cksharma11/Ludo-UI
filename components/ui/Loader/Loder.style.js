import css from 'styled-jsx/css';
import colors from '../../../config/styles/colors';

const LoaderStyles = css`
  .loader {
    border: 10px solid ${colors.gray.shade1};
    border-radius: 50%;
    border-top: 10px solid ${colors.gray.shade3};
    width: 50px;
    height: 50px;
    -webkit-animation: spin 2s linear infinite; /* Safari */
    animation: spin 2s linear infinite;
  }

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export default LoaderStyles;
