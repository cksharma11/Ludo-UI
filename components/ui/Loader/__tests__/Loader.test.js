import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../Loader';

describe('Loader', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
