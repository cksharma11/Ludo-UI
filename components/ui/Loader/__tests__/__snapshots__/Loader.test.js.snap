// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render properly 1`] = `
"<div className=\\"jsx-627297516 loader\\">
  <JSXStyle id=\\"627297516\\">
    .
    l
    o
    a
    d
    e
    r
    .
    j
    s
    x
    -
    6
    2
    7
    2
    9
    7
    5
    1
    6
    {
    b
    o
    r
    d
    e
    r
    :
    1
    6
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    f
    3
    f
    3
    f
    3
    ;
    b
    o
    r
    d
    e
    r
    -
    r
    a
    d
    i
    u
    s
    :
    5
    0
    %
    ;
    b
    o
    r
    d
    e
    r
    -
    t
    o
    p
    :
    1
    6
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    3
    4
    9
    8
    d
    b
    ;
    w
    i
    d
    t
    h
    :
    1
    2
    0
    p
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    1
    2
    0
    p
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    a
    n
    i
    m
    a
    t
    i
    o
    n
    :
    s
    p
    i
    n
     
    2
    s
     
    l
    i
    n
    e
    a
    r
     
    i
    n
    f
    i
    n
    i
    t
    e
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    a
    n
    i
    m
    a
    t
    i
    o
    n
    :
    s
    p
    i
    n
    -
    j
    s
    x
    -
    6
    2
    7
    2
    9
    7
    5
    1
    6
     
    2
    s
     
    l
    i
    n
    e
    a
    r
     
    i
    n
    f
    i
    n
    i
    t
    e
    ;
    a
    n
    i
    m
    a
    t
    i
    o
    n
    :
    s
    p
    i
    n
    -
    j
    s
    x
    -
    6
    2
    7
    2
    9
    7
    5
    1
    6
     
    2
    s
     
    l
    i
    n
    e
    a
    r
     
    i
    n
    f
    i
    n
    i
    t
    e
    ;
    }
    @
    -
    w
    e
    b
    k
    i
    t
    -
    k
    e
    y
    f
    r
    a
    m
    e
    s
     
    s
    p
    i
    n
    -
    j
    s
    x
    -
    6
    2
    7
    2
    9
    7
    5
    1
    6
    {
    0
    %
    {
    -
    w
    e
    b
    k
    i
    t
    -
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    0
    d
    e
    g
    )
    ;
    -
    m
    s
    -
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    0
    d
    e
    g
    )
    ;
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    0
    d
    e
    g
    )
    ;
    }
    1
    0
    0
    %
    {
    -
    w
    e
    b
    k
    i
    t
    -
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    3
    6
    0
    d
    e
    g
    )
    ;
    -
    m
    s
    -
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    3
    6
    0
    d
    e
    g
    )
    ;
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    3
    6
    0
    d
    e
    g
    )
    ;
    }
    }
    @
    k
    e
    y
    f
    r
    a
    m
    e
    s
     
    s
    p
    i
    n
    -
    j
    s
    x
    -
    6
    2
    7
    2
    9
    7
    5
    1
    6
    {
    0
    %
    {
    -
    w
    e
    b
    k
    i
    t
    -
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    0
    d
    e
    g
    )
    ;
    -
    m
    s
    -
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    0
    d
    e
    g
    )
    ;
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    0
    d
    e
    g
    )
    ;
    }
    1
    0
    0
    %
    {
    -
    w
    e
    b
    k
    i
    t
    -
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    3
    6
    0
    d
    e
    g
    )
    ;
    -
    m
    s
    -
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    3
    6
    0
    d
    e
    g
    )
    ;
    t
    r
    a
    n
    s
    f
    o
    r
    m
    :
    r
    o
    t
    a
    t
    e
    (
    3
    6
    0
    d
    e
    g
    )
    ;
    }
    }
  </JSXStyle>
</div>"
`;
